proposals/NNNN-method-keypaths.md

```diff
@@ -15,18 +15,18 @@ expressly limited key paths to be able to reference only properties and subscrip
 \Person.pets[0]               // KeyPath<Person, Pet>
 ```
 
-This proposal adds the ability for key paths to reference methods, optionally specifying argument names:
+This proposal adds the ability for key paths to reference instance methods, optionally specifying argument names to disambiguate:
 
 ```swift
 \Person.sing                  // KeyPath<Person, () -> Sound>
-\Person.sing(melody:lyrics:)  // KeyPath<Person, (Melody,String) -> Sound>
+\Person.sing(melody:lyrics:)  // KeyPath<Person, (Melody, String) -> Sound>
 ```
 
-Note that these key paths do not provide argument values; they reference _unapplied_ methods, and the value they give is
-a closure and not the result of calling the method.
+Note that these key paths do not provide argument values: they reference _unapplied_ methods, and the value they give is
```

Comment:
The semicolon is the grammatically correct punctuation here.

Reply:
The grammatical rules here are extremely obtuse, but I’ve [looked](https://writingcenter.unc.edu/tips-and-tools/semi-colons-colons-and-dashes/) [this](https://leo.stcloudstate.edu/punct/col-semi.html) [up](https://www2.le.ac.uk/offices/ld/all-resources/writing/grammar/grammar-guides/semicolon) to verify them: colons are used when the second clause is an example or explanation of the first.

Reply:
Yes, you can make a good case for the colon too. The semicolon is correct here because the phrase after it is an independent clause. There is not a problem here that needs solving.

The links you gave provide examples where one could construe the second clause to be an example or explanation, but they nonetheless chose to use a semicolon:

>  I bought a ton of fruit; apples, grapes, and pears were all on sale.

> There was no running and no shouting; all the children behaved very well.

---

This kind of editorial feedback isn’t particularly useful. Please limit the grammatical nit-picking to places where there is truly a problem, rather than a difference in taste! The goal of providing edits is not to shift the document into your voice.

Instead, we could use help filling out some of the TODOs. If you have some time, check our [discussion thread](https://forums.swift.org/t/proposal-to-allow-methods-in-key-paths/35444/3) and see what TODOs don’t have somebody working on them already.